Applications/Spire/Include/Spire/KeyBindings/KeyBindingsWindow.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef SPIRE_KEY_BINDINGS_WINDOW_HPP
+#define SPIRE_KEY_BINDINGS_WINDOW_HPP
+#include "Spire/KeyBindings/KeyBindings.hpp"
+#include "Spire/KeyBindings/OrderTasksPage.hpp"
+#include "Spire/Ui/ComboBox.hpp"
+#include "Spire/Ui/Window.hpp"
+
+namespace Spire {
+
+  /** Display the key bindings window. */
+  class KeyBindingsWindow : public Window {
+    public:
+
+      /**
+       * Signals the submission of key bindings of order tasks.
+       * @param submission The order tasks key bindings to be submitted.
+       */
+      using SubmitSignal = Signal<void (
```

Comment:
Why is this optional?

Reply:
It should be removed. 
The submit signal can submit `order tasks`, `cancel tasks` and `interactions` (the last two will be added in the future). My initial thought is that the “Apply” button is page-specific. When the user presses “Apply” on the `OrderTasksPage`, only `order tasks` should be submitted and other two submissions would be none. After talking to Jon later, I know that “Apply” applies all changes that have been made across all pages rather than being page-specific, so this `optional` is no more needed.

Reply:
Thanks for the clarification.